packages/lit-html/src/lit-html.ts

```diff
@@ -755,6 +767,12 @@ class TemplateInstance {
     this._template = template;
   }
 
+  _disconnect() {
+    for (const part of this._parts) {
```

Comment:
Couldn't `TemplateInstance` have a `_hasDisconnect` flag and avoid this work if it's not set?

Reply:
Done